mobile/lib/common/settings_details.dart

```diff
@@ -0,0 +1,183 @@
+                            ),
+                          ],
+                        ),
+                      )
+                    ],
+                  ),
+                  const SizedBox(
+                    height: 20,
+                  ),
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        "Node Info".toUpperCase(),
+                        style:
+                            const TextStyle(color: Colors.grey, fontSize: 17),
+                      ),
+                      const SizedBox(
+                        height: 10,
+                      ),
+                      Container(
+                          decoration: BoxDecoration(
+                              color: Colors.white,
+                              borderRadius: BorderRadius.circular(15)),
+                          child: Column(
+                            children: [
+                              moreInfo(context,
+                                  title: "Node Id", info: widget.nodeId)
+                            ],
+                          ))
+                    ],
+                  ),
+                  Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        "Build Info".toUpperCase(),
+                        style:
+                            const TextStyle(color: Colors.grey, fontSize: 18),
+                      ),
+                      const SizedBox(
+                        height: 10,
+                      ),
+                      Container(
+                          decoration: BoxDecoration(
+                              color: Colors.white,
+                              borderRadius: BorderRadius.circular(15)),
+                          child: Column(
+                            children: [
+                              moreInfo(context,
+                                  title: "Number",
+                                  info: widget.number,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Version",
+                                  info: widget.version,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Commit Hash", info: widget.branch),
+                              moreInfo(context,
+                                  title: "Branch", info: "Main", lessInfo: true)
+                            ],
+                          ))
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          )),
+    );
+  }
+}
+
+Widget moreInfo(BuildContext context,
+    {required String title, required String info, bool lessInfo = false}) {
```

Comment:
The variable `lessInfo` could be more expressive. It's not clear that it actually adds a copy & paste button.